### ISSUE TITLE
Text-Input Cards

### ISSUE THREAD

**Issue by ram535** [l3kn/org-fc#18]
Provide an example of how set a text-input card.
Thanks

**Comment by l3kn** [comment]
Hi,

In one of the last commits, I removed the for text-input card type because I couldn't figure out how to implement a review function for them. I'll update the docs to reflect that change.

How would you like them to work?

My idea was to use the content of the "Back" heading as the expexted answer,
removing surrounding org emphasis markers ("=test=" -> "test", "~test~" -> "test") so the answer can be formatted as verbatim / code in org-mode.

Reviewing them would prompt the user for input,
then either show the answer in green if it was correct or in red (next to the expected answer) if it was wrong.

**Comment by ram535** [comment]
That's the feature that I really miss from anki. It's really useful for practicing spelling when learning a new language. 

> My idea was to use the content of the "Back" heading as the expected answer,
removing surrounding org emphasis markers (=test= -> "test", ~test~ -> "test") so the answer can be formatted as verbatim / code in org-mode.

> Reviewing them would prompt the user for input,
then either show the answer in green if it was correct or in red (next to the expected answer) if it was wrong.

That sounds good. In anki, it only mark in red the part that was misspelled. I think that is a good way to do it. Instead of marking everything in red when an error is made.

![Untitled](https://user-images.githubusercontent.com/9011002/77701644-06d91180-6f74-11ea-8f04-45b274809112.png)

**Comment by l3kn** [comment]
Sorry for the delay, I had some trouble making the diff overlay with emphasized answers.

I've added a new text-input card type that tries to emulate the way text input cards work in Anki,
the only difference being that the expected answer is displayed to the right, not below the user input.

See https://github.com/l3kn/org-fc/blob/master/doc/card_types.org#text-input-cards for documentation on this new card type.

Short version: the first line of a cards content (or its "Back" heading contents) is treated as the answer.

Can you try it out and let me know if it works as expected?

**Comment by ram535** [comment]
Right now I am getting this error when doing `org-fc-dashboard`.
![image](https://user-images.githubusercontent.com/9011002/79083985-4890e880-7ce6-11ea-970b-c2ba1c4c95d9.png)

And this one when doing `org-fc-review`
![image](https://user-images.githubusercontent.com/9011002/79084041-94439200-7ce6-11ea-9d8b-3b9af0fbc78f.png)

I've been modifying a lot my dots lately. I am not sure if it is me provoking those errors somehow.

**Comment by l3kn** [comment]
I think that error comes from org-fc not finding the awk files,
I should add a check for that with better error messages.

You can check the value of `org-fc-source-path` and see if it contains an awk folder.

**Comment by ram535** [comment]
This is the value of `org-fc-source-path` `~/.emacs.d/straight/build/org-fc/`

**Comment by l3kn** [comment]
Weird, given your information from #16, I'd expect it to resolve the symlink and point to `straight/repos/org-fc/`.

Can you do a `ls -l` on `~/.emacs.d/straight/build/org-fc/`? Maybe something has changed with the way straight installs packages?

**Comment by ram535** [comment]
```
-rw-rw-r-- 1 last last  2136 Apr 12 15:24 org-fc-autoloads.el
lrwxrwxrwx 1 last last    51 Apr 12 15:24 org-fc.el -> ~/.emacs.d/straight/repos/org-fc/org-fc.el
-rw-rw-r-- 1 last last 70776 Apr 12 15:24 org-fc.elc
lrwxrwxrwx 1 last last    57 Apr 12 15:24 org-fc-hydra.el -> ~/.emacs.d/straight/repos/org-fc/org-fc-hydra.el
-rw-rw-r-- 1 last last 15742 Apr 12 15:24 org-fc-hydra.elc
```

**Comment by l3kn** [comment]
You can put `(setq org-fc-source-path "~/.emacs.d/straight/repos/org-fc/")` in your config as a quick workaround.

I'll set up straight and investigate further tomorrow.

**Comment by ram535** [comment]
I applied the quick workaround and tested the input card future. It works correctly. Thanks a lot for your work, I appreciate it.

**Comment by l3kn** [comment]
You're welcome, thanks for testing it!

I'll close this issue and open up a new one for the incorrect source path.
